S01E02_BeSquare/SquareFactory.cs

```diff
@@ -36,7 +40,11 @@ public void AddMaterial(int width, int height)
                     var w = Math.Min(SquareSize, remainingWidth);
                     var h = Math.Min(SquareSize, remainingHeight);
 
-                    _ready.Enqueue(new Rectangle(w, h));
+                    // Added a little check here, so that the leftovers from the edges of material are not piled up in the square enqueue.
```

Comment:
I get the idea and it seems to solve the problem but the resulting code is kind of weird. Above you see the `verticalPieces` and `horizontalPieces` variables. Now their values are a lie - a different number of pieces is actually produced by the factory!

Better solution would be to ensure that these variables already have the right values to start with, and that the loop will never try to process a partial piece.

So this is a bit questionable from a readability standpoint (variables with names that lie) and less "if" statements is often better stylistically. It might be correct code, of course, and it is especially nice that you added the comment explaining what is what - even if you leave it as-is, the comment means the next person can at least understand why the two variables lie.